詳細機能実装

sorasora313 opened the pull request:
#What
商品詳細表示機能実装

#Why
商品の詳細をユーザーに見てもらうため


[ログイン状態且つ、自身が出品した販売中商品の商品詳細ページへ遷移した動画](https://gyazo.com/10bedb534817350b76c92f751806da2d)
 [ログイン状態且つ、自身が出品していない販売中商品の商品詳細ページへ遷移した動画](https://gyazo.com/22837e4762af0c5282d6ba388ac09a82)

[ ログアウト状態で、商品詳細ページへ遷移した動画](https://gyazo.com/508f91c9d3fbad4ca5d530d83f693625)

**ブランチを作り忘れたことに気がついて、show.html.erbだけは別のファイルに一旦保存して、貼り付けました。**

sorasora313 committed:
詳細機能実装

ブランチを作り忘れたことに気がついて、show.html.erbだけは別のファイルに一旦保存して、貼り付けました。

sh10n7 commented on app/controllers/items_controller.rb:
```diff
@@ -18,7 +18,9 @@ def create
   end
 
-
+  def show
```
現状ログアウト状態では、詳細ページに遷移できないようになられています。
以下実装条件を満たすため、ログアウト状態でも詳細ページは見れるように記述を修正しましょう!

```
「商品詳細表示ページは、ログイン状況や商品の販売状況に関係なく、誰でも見ることができること。」という実装条件を満たすためです。
```

sh10n7 commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,111 @@
+      
+    <% end %>
+
+    <%# //ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>
```
```
<%# //ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>
```


こちら実装の案内文になりますので、実装が終わりましたら削除しましょう。
実装案内は完成したアプリを人に見せる際に残しておくの望ましくないためです。

このファイル内に他にも実装済みの案内文がないか確認し、あれば削除しましょう！

sh10n7 commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,111 @@
+        ¥ 999,999,999
+      </span>
+      <span class="item-postage">
+        <%= "配送料負担" %>
```
```
<%= "配送料負担" %>
```
配送料負担の記述も、商品画像や商品名と同様、データベースから取得するように記述を修正しましょう！

sh10n7 commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,111 @@
+    </div>
+    <div class="item-price-box">
+      <span class="item-price">
+        ¥ 999,999,999
```
```
¥ 999,999,999
```
商品画像、商品名と同様にこちらもデータベースに登録されている金額が表示されるように、記述を修正しましょう！

sh10n7 commented on app/views/items/show.html.erb:
```diff
@@ -0,0 +1,111 @@
+    </a>
+  </div>
+  <%# 詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
+  <a href="#" class="another-item"><%= "商品のカテゴリー名" %>をもっと見る</a>
```
```
<%= "商品のカテゴリー名" %>
```
こちらもカテゴリー名をデータベースから取得し、表示できるように修正しましょう！

sh10n7 reviewed (commented):
レビュー依頼ありがとうございます。
修正点をコメントをさせていただきましたので、ご確認をお願いいたします。

また、1点以下の点も併せて修正をお願いできますでしょうか？
```
・商品がすでに出品されている場合は、出品されている商品が一覧で表示されること
・商品が出品されていない状態では、ダミーの商品情報が表示されること
```
こちら一覧表示機能で実装いただく箇所となりますが、現状商品出品中でもダミーデータが表示されておりますので、併せて修正をお願いいたします。


また、修正後はコミット・プッシュをご確認いただき、再度**レビュー依頼フォーム**のご提出をお願いいたします。

ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
どうぞ、よろしくお願いいたします。

sorasora313 committed:
修正１

zonodance reviewed (commented):
レビュー依頼ありがとうございます！
お手数ですが、下記の内容も修正していただいた上で動画を撮っていただいた上で、こちらのプルリクエストにて再度ご提出いただきたく存じます。
```
・商品がすでに出品されている場合は、出品されている商品が一覧で表示されること
・商品が出品されていない状態では、ダミーの商品情報が表示されること
こちら一覧表示機能で実装いただく箇所となりますが、現状商品出品中でもダミーデータが表示されておりますので、併せて修正をお願いいたします。
```
ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
どうぞ、よろしくお願いいたします。

sorasora313 commented:
[商品がすでに出品されている場合は、出品されている商品が一覧で表示されること](https://gyazo.com/ad61f76ab0df7afc3666585040526c49)
[商品が出品されていない状態では、ダミーの商品情報が表示されること](https://gyazo.com/5e2d9c367e04a0c09b20f7764165511f)

zonodance reviewed (commented):
レビュー依頼、ありがとうございます！動画のご提出もありがとうございます。
こちらの[Gyazo](https://gyazo.com/ad61f76ab0df7afc3666585040526c49)を拝見すると、
現在、商品が出品されている場合もダミー画像が出ている状況になっております。
商品が出品されている場合には、ダミー画像が出ないようにしてみましょう。

sorasora313 committed:
修正しました

[・商品がすでに出品されている場合は、出品されている商品が一覧で表示されること](https://gyazo.com/4a4c8dd751c9b45ff9543c04c7f0d9c2)
[・商品が出品されていない状態では、ダミーの商品情報が表示されること](https://gyazo.com/4fd76645c572d1cb5ba1681b875e5c0e)

zonodance reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
アプリが動くかの確認も必要となりますので、下記の挙動確認の動画の提出もお願いいたします。

・商品がすでに出品されている場合は、出品されている商品が一覧で表示されること
・商品が出品されていない状態では、ダミーの商品情報が表示されること

お手数をおかけしますが、正確なレビューのためご協力いただければ幸いです。
よろしくお願いいたします。

sorasora313 commented:
[・商品がすでに出品されている場合は、出品されている商品が一覧で表示されること](https://gyazo.com/4a4c8dd751c9b45ff9543c04c7f0d9c2)
[・商品が出品されていない状態では、ダミーの商品情報が表示されること](https://gyazo.com/4fd76645c572d1cb5ba1681b875e5c0e)

zonodance reviewed (commented):
レビュー依頼ありがとうございます！

`LGTM`とさせていただきます！商品詳細表示機能の実装お疲れさまでした。
次のブランチを切る前にMergeとPullを忘れないようお願いします！
